backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -222,56 +222,72 @@ public LiveOktaRepository(
     User user = users.single();
 
     String orgId = org.getExternalId();
-    boolean userInOrg = false;
-    // Remove user from old groups
-    for (Group g : user.listGroups()) {
-      if (g.getType() == GroupType.OKTA_GROUP
-          && g.getProfile().getName().startsWith(generateGroupOrgPrefix(orgId))) {
-        userInOrg = true;
-        // do not remove user from org's default group
-        if (!g.getProfile()
-            .getName()
-            .equals(generateRoleGroupName(orgId, OrganizationRole.getDefault()))) {
-          g.removeUser(user.getId());
-        }
-      }
-    }
-    if (!userInOrg) {
+
+    final String groupOrgPrefix = generateGroupOrgPrefix(orgId);
+    final String groupOrgDefaultName = generateRoleGroupName(orgId, OrganizationRole.getDefault());
+
+    Map<String, Group> currentOrgGroupMapForUser =
+        user.listGroups().stream()
+            .filter(
+                g ->
+                    GroupType.OKTA_GROUP.equals(g.getType())
+                        && g.getProfile().getName().startsWith(groupOrgPrefix))
+            .collect(Collectors.toMap(g -> g.getProfile().getName(), g -> g));
+
+    if (!currentOrgGroupMapForUser.containsKey(groupOrgDefaultName)) {
+      currentOrgGroupMapForUser.values().forEach(g -> g.removeUser(user.getId()));
```

Comment:
It seems like this is clearing `currentOrgGroupMapForUser`, but why is that necessary since we're throwing an exception?

Reply:
This is to retain the behavior the method previously had.  It covers the case where the user had some organizational group memberships, but was not a member of their `NO_ACCESS` group.  Since that is an invalid configuration, this will purge their remaining org-level group memberships.

Good point, it isn't obvious why this is done.